storage/badger/store.go

```
@@ -21,43 +21,236 @@ package badger
 	"errors"
 	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
 
 	"github.com/dgraph-io/badger/v2"
-	"github.com/onflow/flow-go/engine/execution/state/delta"
-	flowgo "github.com/onflow/flow-go/model/flow"
-
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/google/uuid"
 	"github.com/onflow/flow-emulator/storage"
 	"github.com/onflow/flow-emulator/types"
+	"github.com/onflow/flow-go/engine/execution/state/delta"
+	flowgo "github.com/onflow/flow-go/model/flow"
 )
 
 // Store is an embedded storage implementation using Badger as the underlying
 // persistent key-value store.
 type Store struct {
 	db              *badger.DB
 	ledgerChangeLog changelog
+	dbGitRepository *git.Repository
+	path            string
+	badgerOptions   badger.Options
 }
 
 var _ storage.Store = &Store{}
 
+func getTag(r *git.Repository, tag string) *object.Tag {
+	tags, err := r.TagObjects()
+	if err != nil {
+		return nil
+	}
+	var res *object.Tag = nil
+	tags.ForEach(func(t *object.Tag) error {
+		if t.Name == tag {
+			res = t
+		}
+		return nil
+	})
+	return res
+}
+
+func setTag(r *git.Repository, tag string, tagger *object.Signature) (bool, error) {
+	if getTag(r, tag) != nil {
+		return false, nil
+	}
+	h, err := r.Head()
+	if err != nil {
+		return false, err
+	}
+	_, err = r.CreateTag(tag, h.Hash(), &git.CreateTagOptions{
+		Tagger:  tagger,
+		Message: tag,
+	})
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+func defaultSignature(name, email string) *object.Signature {
+	return &object.Signature{
+		Name:  name,
+		Email: email,
+		When:  time.Now(),
+	}
+}
+
+//prevents git commits when emulator running
+func (s *Store) lockGit() {
+	lockPath := fmt.Sprintf("%s/.git/index.lock", s.path)
+	ioutil.WriteFile(lockPath, []byte("emulatorLock"), 0755)
+}
+
+func (s *Store) unlockGit() {
+	lockPath := fmt.Sprintf("%s/.git/index.lock", s.path)
+	os.Remove(lockPath)
```

Explicity ignore error if you are sure there is no need to handle it